### PR TITLE
Dockerization

### PR DESCRIPTION
**Pull request by git-voo**

### CODE REVIEW

File: .github/workflows/cd_prod.yml
Comment (mohamdlog) [inline .github/workflows/cd_prod.yml:1]: Addition of Dockerfile and docker-compose.yml to containerize the application.

File: backend/models/conversationsModel.js
Comment (mohamdlog) [inline backend/models/conversationsModel.js:1]: Creation of a Conversation model and corresponding routes to manage user conversations.

File: backend/server.js
Comment (mohamdlog) [inline backend/server.js:1]: Implementation of a WebSocket server using socket.io to facilitate real-time messaging.

**Review by mohamdlog — APPROVED** [review]
This PR introduces essential updates, including Dockerization with Dockerfile and docker-compose.yml for seamless deployment, real-time messaging with WebSocket integration in the backend, and a new Conversations component on the frontend to enhance user interaction. The addition of Google Drive integration tests ensures robust file handling, while project cleanup and dependency updates improve maintainability. These changes significantly enhance the app's functionality, scalability, and user experience. Approving this PR as it aligns with the project’s goals and is well-implemented.

**Review by shagunsharmaslu — APPROVED** [review]
Docker integration is enabling seamless and consistent deployment across environments. It also adds a new `Conversations` component, leveraging WebSockets for real-time messaging, enhancing communication between users. The frontend has been updated to support these features, ensuring a smooth user experience. These updates significantly improve the platform's functionality, deployment efficiency, and user engagement.

**Pull request merged**